### PR TITLE
WIP Change static class to bean and fix default port bug

### PR DESCRIPTION
**Pull request by khotyn**
Change the static classes to Spring Bean and fix the default port bugs

The major change of this pull request is to change the code more Spring way, but according to the Spring way, see we have the following configuration

```java
@ConfigurationProperties("com.alipay.sofa.rpc")
public class SofaBootRpcProperties {
    private String boltPort;
}
```

Spring Boot will change the properties to `com.alipay.sofa.rpc.bolt-port` but not `com.alipaysofa.rpc.bolt.port`, this will introduce compatible problem with the current version.

How you guys think of such problem? Should we hack some code to maintain the compatible with current version and leverage the convenience of Spring Boot?

@ujjboy @NeGnail @leizhiyuan @guanchao-yang

### CODE REVIEW

**Comment by ujjboy** [comment]
`com.alipaysofa.rpc.bolt.port` is better than `com.alipay.sofa.rpc.bolt-port`. 

How about try like this:
```
@ConfigurationProperties("com.alipay.sofa.rpc")
public class SofaBootRpcProperties {
    private BoltProperties bolt;
}
public class BoltProperties {
    private int port;
}
```

File: sofa-boot-starter/src/main/java/com/alipay/sofa/rpc/boot/container/ServerConfigContainer.java [hunk 1]
```diff
@@ -135,30 +143,31 @@ public static ServerConfig getServerConfig(String protocol) {
...
+        String ioThreadCountStr = sofaBootRpcProperties.getBoltIoThreadCount();
+        String executorThreadCountStr = sofaBootRpcProperties.getBoltExecutorThreadCount();
+        String acceptsCountStr = sofaBootRpcProperties.getBoltAcceptsCount();
 
         ServerConfig serverConfig = new ServerConfig();
 
         if (StringUtils.hasText(portStr)) {
             serverConfig.setPort(Integer.parseInt(portStr));
+        } else {
+            serverConfig.setPort(12200);
```
Comment (ujjboy) [inline sofa-boot-starter/src/main/java/com/alipay/sofa/rpc/boot/container/ServerConfigContainer.java]: magic value.  need add to `SofaBootRpcConfigConstants`

**Comment by khotyn** [comment]
@ujjboy Just tried your way, need a looooot of class to implement like this.

**Comment by khotyn** [comment]
@ujjboy Implement a hacky way to get the dot way works:

```java
public String getAftRegulationEffective() {
    return StringUtils.isEmpty(aftRegulationEffective) ? getDotString(new Object() {
    }.getClass().getEnclosingMethod().getName()) : aftRegulationEffective;
}
```

But the IDE still hint the Spring way:

image

I think this is OK, user can use dot or Spring way as their favorite.

**Comment by NeGnail** [comment]
```java
@ConfigurationProperties("com.alipay.sofa.rpc")
public class SofaBootRpcProperties {
    private String boltPort;
}
```
I'm trying to write a demo, and I can get value through `com.alipay.sofa.rpc.bolt.port.`

File: sofa-boot-starter/pom.xml [hunk 2]
```diff
@@ -304,6 +303,7 @@
                     <destFile>${jacoco.path}</destFile>
                     <dataFile>${jacoco.path}</dataFile>
                     <sessionId>jacoco_coverage</sessionId>
+                    <append>true</append>
                     <excludes>
```
Comment (ujjboy) [inline sofa-boot-starter/pom.xml]: @NeGnail Is this `excludes` necessarily?

Comment (NeGnail) [reply sofa-boot-starter/pom.xml]: @ujjboy    it is unnecessary.  my next pr will delete it.

**Pull request merged**